lib/streamlit/caching.py

```diff
@@ -53,6 +54,18 @@
 to suppress the warning.
 """
 
+# based on: https://stackoverflow.com/questions/43506378/how-to-get-source-code-of-function-that-is-wrapped-by-a-decorator
+# In Python 2, the @functools.wraps() decorator does not set the convenience __wrapped__
+if sys.version_info[0:2] >= (3, 4):  # Python v3.4+?
```

Comment:
Looks like it might be 3.2

> New in version 3.2: Automatic addition of the __wrapped__ attribute.

> To allow access to the original function for introspection and other purposes (e.g. bypassing a caching decorator such as lru_cache()), this function automatically adds a __wrapped__ attribute to the wrapper that refers to the original function.

> Changed in version 3.4: The __wrapped__ attribute now always refers to the wrapped function, even if that function defined a __wrapped__ attribute. (see issue 17482)

Reply:
Oh I just read the comment on stackoverflow, so I see why you chose 3.4